aten/src/ATen/native/xnnpack/Convolution.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#ifdef USE_XNNPACK
+
+#include <ATen/Tensor.h>
+#include <ATen/native/xnnpack/Common.h>
+#include <ATen/native/xnnpack/OpContext.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+namespace convolution2d {
+
+class Conv2dPrePack final : public torch::OperatorKernel {
+  public:
+  c10::intrusive_ptr<xnnpack::XNNPackConv2dOpContext> operator()(
+      Tensor weight,
+      c10::optional<Tensor> bias,
+      std::vector<int64_t> padding,
+      std::vector<int64_t> stride,
+      std::vector<int64_t> dilation,
+      int64_t groups);
+};
+
+class Conv2dPacked final : public torch::OperatorKernel {
+  public:
+  Tensor operator()(
+      const Tensor& input,
+      const c10::intrusive_ptr<xnnpack::XNNPackConv2dOpContext>& op_context);
+};
+
+Context create(
+    const Tensor& weight,
+    const c10::optional<Tensor>& bias,
+    const IntArrayRef padding_,
+    const IntArrayRef stride_,
+    const IntArrayRef dilation_,
```

Comment:
Why do these have trailing underscores?

Reply:
Thanks for pointing this out. I had just moved these functions around, so it was there before. I will remote trailing _.